### ISSUE TITLE
widget buttons not working

### ISSUE THREAD

**Issue by Micha-Btz** [SimpleMobileTools/Simple-Calendar#240]
Hey,
I have version 2.8.1 from f-droid an the button to switch the displayed month and the plus button to add a new date are not working.

```
10-17 09:21:32.896 W/ResourceType(1284): No package identifier when getting value for resource number 0x00000000
10-17 09:21:32.905 W/PackageManager(1284): Failure retrieving resources for com.nolanlawson.logcat: Resource ID #0x0
10-17 09:21:32.949 I/ActivityManager(840): START u0 {act=com.android.systemui.recents.SHOW_RECENTS flg=0x10804000 cmp=com.android.systemui/.recents.RecentsActivity} from uid 10016 on display 0
10-17 09:21:33.291 D/audio_hw_primary(307): No space available in compress driver, post msg to cb thread
10-17 09:21:33.357 I/Timeline(1284): Timeline: Activity_idle id: android.os.BinderProxy@31f95c02 time:38708573
10-17 09:21:33.508 I/Timeline(840): Timeline: Activity_windows_visible id: ActivityRecord{42b12e6 u0 com.android.systemui/.recents.RecentsActivity t8556} time:38708724
10-17 09:21:34.012 W/ResourceType(1284): No package identifier when getting value for resource number 0x00000000
10-17 09:21:34.012 W/PackageManager(1284): Failure retrieving resources for com.nolanlawson.logcat: Resource ID #0x0
10-17 09:21:34.303 D/audio_hw_primary(307): No space available in compress driver, post msg to cb thread
10-17 09:21:35.335 D/audio_hw_primary(307): No space available in compress driver, post msg to cb thread
10-17 09:21:35.546 I/ActivityManager(840): START u0 {flg=0x10100000 cmp=com.simplemobiletools.calendar/.activities.SplashActivity} from uid 10003 on display 0
10-17 09:21:35.561 V/WindowManager(840): addAppToken: AppWindowToken{bd0cb token=Token{367e5f9a ActivityRecord{5fbc45 u0 com.simplemobiletools.calendar/.activities.SplashActivity t8628}}} to stack=1 task=8628 at 0
10-17 09:21:35.615 I/Timeline(18582): Timeline: Activity_launch_request id:com.simplemobiletools.calendar time:38710831
10-17 09:21:35.616 I/ActivityManager(840): START u0 {cmp=com.simplemobiletools.calendar/.activities.MainActivity} from uid 10003 on display 0
10-17 09:21:35.622 V/WindowManager(840): addAppToken: AppWindowToken{fd2db66 token=Token{2ac4d7c1 ActivityRecord{26a7f0a8 u0 com.simplemobiletools.calendar/.activities.MainActivity t8628}}} to stack=1 task=8628 at 1
10-17 09:21:35.653 V/WindowManager(840): rotationForOrientationLw(orient=1, last=0); user=0 
10-17 09:21:35.665 V/WindowManager(840): Based on layer: Adding window Window{bdf6f54 u0 com.simplemobiletools.calendar/com.simplemobiletools.calendar.activities.MainActivity} at 20 of 26
10-17 09:21:36.045 I/ActivityManager(840): Displayed com.simplemobiletools.calendar/.activities.MainActivity: +416ms (total +476ms)
10-17 09:21:36.132 I/Timeline(18582): Timeline: Activity_idle id: android.os.BinderProxy@3004c019 time:38711348
10-17 09:21:36.361 D/audio_hw_primary(307): No space available in compress driver, post msg to cb thread
10-17 09:21:36.407 I/Timeline(840): Timeline: Activity_windows_visible id: ActivityRecord{26a7f0a8 u0 com.simplemobiletools.calendar/.activities.MainActivity t8628} time:38711623
10-17 09:21:37.153 I/ActivityManager(840): START u0 {act=android.intent.action.MAIN cat=[android.intent.category.HOME] flg=0x10200000 cmp=com.cyanogenmod.trebuchet/com.android.launcher3.Launcher} from uid 1000 on display 0
10-17 09:21:37.168 V/WindowManager(840): rotationForOrientationLw(orient=5, last=0); user=0 
10-17 09:21:37.234 I/Timeline(1677): Timeline: Activity_idle id: android.os.BinderProxy@3caa356 time:38712450
10-17 09:21:37.373 D/audio_hw_primary(307): No space available in compress driver, post msg to cb thread
10-17 09:21:37.492 I/Timeline(840): Timeline: Activity_windows_visible id: ActivityRecord{32601a75 u0 com.cyanogenmod.trebuchet/com.android.launcher3.Launcher t8550} time:38712708
10-17 09:21:37.736 W/OpenGLRenderer(1677): Incorrectly called buildLayer on View: ShortcutAndWidgetContainer, destroying layer...
10-17 09:21:37.737 W/OpenGLRenderer(1677): Incorrectly called buildLayer on View: ShortcutAndWidgetContainer, destroying layer...
10-17 09:21:38.183 W/ResourceType(1284): No package identifier when getting value for resource number 0x00000000
10-17 09:21:38.183 W/PackageManager(1284): Failure retrieving resources for com.simplemobiletools.calendar: Resource ID #0x0
10-17 09:21:42.840 I/ActivityManager(840): START u0 {flg=0x10000000 cmp=com.simplemobiletools.calendar/.activities.SplashActivity bnds=[137,232][354,360]} from uid 10003 on display 0
10-17 09:21:42.964 V/WindowManager(840): rotationForOrientationLw(orient=1, last=0); user=0 
10-17 09:21:43.174 I/Timeline(18582): Timeline: Activity_idle id: android.os.BinderProxy@3004c019 time:38718390
10-17 09:21:43.185 D/OpenGLRenderer(1677): endAllStagingAnimators on 0xa0c88080 (RippleDrawable) with handle 0x9d007390
10-17 09:21:43.452 I/Timeline(840): Timeline: Activity_windows_visible id: ActivityRecord{26a7f0a8 u0 com.simplemobiletools.calendar/.activities.MainActivity t8628} time:38718668
10-17 09:21:43.513 D/audio_hw_primary(307): No space available in compress driver, post msg to cb thread
10-17 09:21:43.773 V/WindowManager(840): rotationForOrientationLw(orient=1, last=0); user=0 
10-17 09:21:44.551 D/audio_hw_primary(307): No space available in compress driver, post msg to cb thread
10-17 09:21:44.717 V/WindowManager(840): rotationForOrientationLw(orient=5, last=0); user=0 
10-17 09:21:44.773 I/Timeline(1677): Timeline: Activity_idle id: android.os.BinderProxy@3caa356 time:38719989
10-17 09:21:45.046 I/Timeline(840): Timeline: Activity_windows_visible id: ActivityRecord{32601a75 u0 com.cyanogenmod.trebuchet/com.android.launcher3.Launcher t8550} time:38720262
10-17 09:21:45.072 V/WindowManager(840): rotationForOrientationLw(orient=5, last=0); user=0 
10-17 09:21:45.288 W/OpenGLRenderer(1677): Incorrectly called buildLayer on View: ShortcutAndWidgetContainer, destroying layer...
10-17 09:21:45.289 W/OpenGLRenderer(1677): Incorrectly called buildLayer on View: ShortcutAndWidgetContainer, destroying layer...
10-17 09:21:45.581 D/audio_hw_primary(307): No space available in compress driver, post msg to cb thread
10-17 09:21:45.731 W/ResourceType(1284): No package identifier when getting value for resource number 0x00000000
10-17 09:21:45.731 W/PackageManager(1284): Failure retrieving resources for com.simplemobiletools.calendar: Resource ID #0x0
10-17 09:21:46.592 D/audio_hw_primary(307): No space available in compress driver, post msg to cb thread
10-17 09:21:46.957 V/WindowManager(840): rotationForOrientationLw(orient=5, last=0); user=0 
10-17 09:21:46.981 I/ActivityManager(840): START u0 {flg=0x10000000 cmp=com.simplemobiletools.calendar/.activities.SplashActivity bnds=[137,232][354,360]} from uid 10003 on display 0
10-17 09:21:47.002 V/WindowManager(840): addAppToken: AppWindowToken{16208f97 token=Token{13eea16 ActivityRecord{19123331 u0 com.simplemobiletools.calendar/.activities.SplashActivity t8629}}} to stack=1 task=8629 at 0
10-17 09:21:47.006 V/WindowManager(840): Adding window Window{232125ee u0 Starting com.simplemobiletools.calendar} at 20 of 26 (after Window{dd1bc69 u0 com.nolanlawson.logcat/com.nolanlawson.logcat.LogcatActivity})
10-17 09:21:47.014 V/WindowManager(840): rotationForOrientationLw(orient=-1, last=0); user=0 
10-17 09:21:47.030 I/Timeline(18582): Timeline: Activity_launch_request id:com.simplemobiletools.calendar time:38722246
10-17 09:21:47.030 I/ActivityManager(840): START u0 {cmp=com.simplemobiletools.calendar/.activities.MainActivity} from uid 10003 on display 0
10-17 09:21:47.042 V/WindowManager(840): addAppToken: AppWindowToken{1a84c125 token=Token{9e1d11c ActivityRecord{12bfda8f u0 com.simplemobiletools.calendar/.activities.MainActivity t8629}}} to stack=1 task=8629 at 1
10-17 09:21:47.042 V/WindowManager(840): Based on layer: Adding window Window{232125ee u0 Starting com.simplemobiletools.calendar} at 20 of 26
10-17 09:21:47.064 D/OpenGLRenderer(1677): endAllStagingAnimators on 0xa0c88080 (RippleDrawable) with handle 0x9d007500
10-17 09:21:47.081 V/WindowManager(840): rotationForOrientationLw(orient=1, last=0); user=0 
10-17 09:21:47.091 V/WindowManager(840): Adding window Window{33c8e308 u0 com.simplemobiletools.calendar/com.simplemobiletools.calendar.activities.MainActivity} at 20 of 27 (before Window{232125ee u0 Starting com.simplemobiletools.calendar})
10-17 09:21:47.464 I/ActivityManager(840): Displayed com.simplemobiletools.calendar/.activities.MainActivity: +407ms (total +450ms)
10-17 09:21:47.464 I/Timeline(840): Timeline: Activity_windows_visible id: ActivityRecord{12bfda8f u0 com.simplemobiletools.calendar/.activities.MainActivity t8629} time:38722680
10-17 09:21:47.558 I/Timeline(18582): Timeline: Activity_idle id: android.os.BinderProxy@110735ff time:38722774
10-17 09:21:47.621 D/audio_hw_primary(307): No space available in compress driver, post msg to cb thread
10-17 09:21:48.222 W/ResourceType(1284): No package identifier when getting value for resource number 0x00000000
10-17 09:21:48.222 W/PackageManager(1284): Failure retrieving resources for com.simplemobiletools.calendar: Resource ID #0x0
10-17 09:21:48.239 I/ActivityManager(840): START u0 {act=com.android.systemui.recents.SHOW_RECENTS flg=0x10804000 cmp=com.android.systemui/.recents.RecentsActivity} from uid 10016 on display 0
10-17 09:21:48.264 V/WindowManager(840): rotationForOrientationLw(orient=-1, last=0); user=0 
10-17 09:21:48.341 I/Timeline(1284): Timeline: Activity_idle id: android.os.BinderProxy@31f95c02 time:38723557
10-17 09:21:48.363 W/InputMethodManagerService(840): Starting input on non-focused client com.android.internal.view.IInputMethodClient$Stub$Proxy@aa13d95 (uid=10003 pid=18582)
10-17 09:21:48.651 D/audio_hw_primary(307): No space available in compress driver, post msg to cb thread
10-17 09:21:48.709 I/Timeline(840): Timeline: Activity_windows_visible id: ActivityRecord{42b12e6 u0 com.android.systemui/.recents.RecentsActivity t8556} time:38723925
10-17 09:21:49.246 V/PhoneStatusBar(1284): setLightsOn(true)
10-17 09:21:49.298 V/PhoneStatusBar(1284): setLightsOn(true)
10-17 09:21:49.369 I/Timeline(24596): Timeline: Activity_idle id: android.os.BinderProxy@2daeb5d2 time:38724585
10-17 09:21:49.651 I/Timeline(840): Timeline: Activity_windows_visible id: ActivityRecord{15f22f9 u0 com.nolanlawson.logcat/.LogcatActivity t8591} time:38724867
10-17 09:21:49.662 D/audio_hw_primary(307): No space available in compress driver, post msg to cb thread
10-17 09:21:49.773 I/art     (840): Background sticky concurrent mark sweep GC freed 106372(5MB) AllocSpace objects, 4(64KB) LOS objects, 11% free, 34MB/39MB, paused 7.552ms total 55.966ms
10-17 09:21:50.108 D/TaskPersister(840): removeObsoleteFile: deleting file=8628_task.xml
10-17 09:21:50.108 D/TaskPersister(840): removeObsoleteFile: deleting file=8628_task_thumbnail.png
10-17 09:21:50.244 I/ActivityManager(840): Process com.cyanogenmod.lockclock (pid 7395) has died
10-17 09:21:50.244 D/ActivityManager(840): cleanUpApplicationRecord -- 7395
10-17 09:21:50.268 W/ResourceType(1284): No package identifier when getting value for resource number 0x00000000
10-17 09:21:50.268 W/PackageManager(1284): Failure retrieving resources for com.simplemobiletools.calendar: Resource ID #0x0
10-17 09:21:50.364 I/art     (24596): Background sticky concurrent mark sweep GC freed 44891(2MB) AllocSpace objects, 0(0B) LOS objects, 17% free, 11MB/13MB, paused 9.448ms total 31.496ms
```
Downgrade to version 2.7.6 solves this.
cm-12.1
Android 5.1.1

**Comment by tibbi** [comment]
Hey, it should be fixed in 2.8.2, which is not available on f-droid yet tho. I plan rewriting the widget handling anyway, the current solution is pretty bad.

**Comment by Micha-Btz** [comment]
update has published to f-droid. Now it works again. Thanks.
